### ISSUE TITLE
NotImplementedError: igemmlt not available (probably built with NO_CUBLASLT)

### ISSUE THREAD

**Issue by stevenmanton** [TimDettmers/bitsandbytes#1242]
### System Info

p3.2xlarge, python 3.12

accelerate==0.30.1
bitsandbytes==0.43.1
transformers==4.40.2
peft==0.10.0

### Reproduction

```python
import transformers
from peft import LoraConfig, TaskType, get_peft_model, prepare_model_for_kbit_training
from transformers import InstructBlipForConditionalGeneration

model = InstructBlipForConditionalGeneration.from_pretrained(
    "Salesforce/instructblip-flan-t5-xl",
    quantization_config=transformers.BitsAndBytesConfig(
        load_in_8bit=True,
    ),
    low_cpu_mem_usage=True,
)
peft_config = LoraConfig(
    task_type=TaskType.FEATURE_EXTRACTION,
    inference_mode=False,
    r=8,
    lora_alpha=32,
    lora_dropout=0.1,
    target_modules="all-linear",
)
model = get_peft_model(
    prepare_model_for_kbit_training(model), peft_config
)
model.merge_and_unload()
```

### Expected behavior

No error. But I see the following stacktrace:

```
NotImplementedError                       Traceback (most recent call last)
Cell In[8], line 1
----> 1 model.merge_and_unload()

File ~/.local/share/hatch/env/pip-compile/amzn-product-dna-science-devel/t-spEv9X/antonstv/lib/python3.12/site-packages/peft/tuners/lora/model.py:784, in LoraModel.merge_and_unload(self, progressbar, safe_merge, adapter_names)
    756 def merge_and_unload(
    757     self, progressbar: bool = False, safe_merge: bool = False, adapter_names: Optional[list[str]] = None
    758 ) -> torch.nn.Module:
    759     r"""
    760     This method merges the LoRa layers into the base model. This is needed if someone wants to use the base model
    761     as a standalone model.
   (...)
    782     ```
    783     """
--> 784     return self._unload_and_optionally_merge(
    785         progressbar=progressbar, safe_merge=safe_merge, adapter_names=adapter_names
    786     )

File ~/.local/share/hatch/env/pip-compile/amzn-product-dna-science-devel/t-spEv9X/antonstv/lib/python3.12/site-packages/peft/tuners/lora/model.py:438, in LoraModel._unload_and_optionally_merge(self, merge, progressbar, safe_merge, adapter_names)
    436 if hasattr(target, "base_layer"):
    437     if merge:
--> 438         target.merge(safe_merge=safe_merge, adapter_names=adapter_names)
    439     self._replace_module(parent, target_name, target.get_base_layer(), target)
    440 elif isinstance(target, ModulesToSaveWrapper):
    441     # save any additional trainable modules part of `modules_to_save`

File ~/.local/share/hatch/env/pip-compile/amzn-product-dna-science-devel/t-spEv9X/antonstv/lib/python3.12/site-packages/peft/tuners/lora/bnb.py:95, in Linear8bitLt.merge(self, safe_merge, adapter_names)
     91     state.SCB = weight.SCB
     93 # Dequantize the result of identity matrix and int8 weight because bitsandbytes does not support int8
     94 # dequantization directly
---> 95 output = dequantize_bnb_weight(weight, state=state)
     96 if not self.use_dora[active_adapter]:
     97     w_data = output.to(lora_data.dtype).to(lora_data.device) + lora_data

File ~/.local/share/hatch/env/pip-compile/amzn-product-dna-science-devel/t-spEv9X/antonstv/lib/python3.12/site-packages/peft/utils/integrations.py:68, in dequantize_bnb_weight(weight, state)
     66 if state.CxB is None:
     67     state.CxB, state.SB = bnb.functional.transform(weight.data, to_order=state.formatB)
---> 68 out32, Sout32 = bnb.functional.igemmlt(im, state.CxB, Sim, state.SB)
     69 return bnb.functional.mm_dequant(out32, Sout32, SCim, state.SCB, bias=None).t()

File ~/.local/share/hatch/env/pip-compile/amzn-product-dna-science-devel/t-spEv9X/antonstv/lib/python3.12/site-packages/bitsandbytes/functional.py:2333, in igemmlt(A, B, SA, SB, out, Sout, dtype)
   2330         has_error = lib.cigemmlt_ampere_8(ptr, m, n, k, ptrA, ptrB, ptrC, ptrRowScale, lda, ldb, ldc)
   2332 if has_error == 100:  # `ERR_NOT_IMPLEMENTED` is defined as 100 in `ops.cu`
-> 2333     raise NotImplementedError("igemmlt not available (probably built with NO_CUBLASLT)")
   2335 if has_error:
   2336     print(f"A: {shapeA}, B: {shapeB}, C: {Sout[0]}; (lda, ldb, ldc): {(lda, ldb, ldc)}; (m, n, k): {(m, n, k)}")

NotImplementedError: igemmlt not available (probably built with NO_CUBLASLT)
```

**Comment by matthewdouglas** [comment]
Hi @stevenmanton,

Thanks for the details. This issue is showing up because the Tesla V100 does not support the int8 tensor cores. There's probably some improvement we can make here on the PEFT and/or BNB side to make this fail more graciously (or fall back to the slower algorithm).

In the meantime, you might have better luck using the 4bit quantization, or switch to an instance with a newer GPU (e.g. g4dn, g5, g6).

**Comment by stevenmanton** [comment]
Thanks for the feedback! Yes, a more descriptive error code would certainly be useful. In my case, I actually trained a model with the 8-bit quantization so that the memory consumption was less. However, I found that the inference was much faster loading the un-quantized model, since I couldn't merge the weights. I saw some slight numerical differences in the output, but nothing very large for the test samples I looked at.
